File: src/backend/domain/object.go
package domain

import "time"

type Object struct {
	Key          string    `json:"key"`
	Size         int64     `json:"size"`
	LastModified time.Time `json:"last_modified"`
	ETag         string    `json:"etag"`
}

type ListObjectsResult struct {
	Objects                 []Object `json:"objects"`
	Prefix                  string   `json:"prefix"`
	Delimiter               string   `json:"delimiter"`
	IsTruncated             bool     `json:"is_truncated"`
	NextContinuationToken   string   `json:"next_continuation_token,omitempty"`
}

File: src/frontend/types/object.ts
export type R2Object = {
  key: string
  size: number
  last_modified: string
  etag: string
}

export type ListObjectsResponse = {
  objects: R2Object[]
  prefix: string
  delimiter: string
  is_truncated: boolean
  next_continuation_token?: string
}

export type DisplayObject = {
  name: string
  key: string
  isFolder: boolean
  size?: number
  lastModified?: string
}
